bijection-core/src/test/scala/com/twitter/bijection/CollectionLaws.scala

```diff
@@ -22,26 +22,50 @@ import org.scalatest.MustMatchers
 class CollectionLaws extends CheckProperties with BaseProperties {
   import com.twitter.bijection.StringArbs._
 
-  implicit def vectorArb[A](implicit la: Arbitrary[List[A]]) =
-    arbitraryViaFn { (l: List[A]) =>
-      Vector(l: _*)
-    }
-  implicit def seqArb[A](implicit la: Arbitrary[List[A]]) =
-    arbitraryViaFn { (l: List[A]) =>
-      Seq(l: _*)
-    }
-  implicit def indexedSeqArb[A](implicit la: Arbitrary[List[A]]) =
-    arbitraryViaFn { (l: List[A]) =>
-      IndexedSeq(l: _*)
-    }
-  implicit def traversableArb[A](implicit la: Arbitrary[List[A]]) =
-    arbitraryViaFn { (l: List[A]) =>
-      l.toTraversable
-    }
+  implicit def vectorArb[A](implicit la: Arbitrary[List[A]]): Arbitrary[Vector[A]] =
+    arbitraryViaFn[List[A], Vector[A]]((l: List[A]) => Vector(l: _*))
+
+  implicit def seqArb[A](implicit la: Arbitrary[List[A]]): Arbitrary[Seq[A]] =
+    arbitraryViaFn[List[A], Seq[A]]((l: List[A]) => Seq(l: _*))
+  
+  implicit def indexedSeqArb[A](implicit la: Arbitrary[List[A]]): Arbitrary[IndexedSeq[A]] =
+    arbitraryViaFn[List[A], IndexedSeq[A]]((l: List[A]) => IndexedSeq(l: _*))
+  
+  implicit def traversableArb[A](implicit la: Arbitrary[List[A]]): Arbitrary[Traversable[A]] =
+    arbitraryViaFn[List[A], Traversable[A]]((l: List[A]) => l.toTraversable)
+
 
   property("round trip List[Int] <=> Vector[String @@ Rep[Int]]") {
-    isBijection[List[Int], Vector[String @@ Rep[Int]]]
-  }
+    type A = Int
+    type B = String @@ Rep[Int]
+    type CA = List[A]
+    type CB = Vector[B]
+
+    val ali = implicitly[Arbitrary[CA]]
+    val avsri = vectorArb(implicitly[Arbitrary[List[B]]])
+    //if we start here with Bijection.fromInjection(Injection.int2String)
+    //things do work out. Without, we get diverging expansion starting at 
+    //Injection.fromImplicitBijection
+    implicit val i1 = Bijection.fromInjection(Injection.int2String)
+    def ibij: ImplicitBijection[CA, CB] = implicitly[ImplicitBijection[List[Int], Vector[String @@ Rep[Int]]]] 
```

Comment:
@johnynek I pushed my WIP, as I'm hitting a hurdle here: scala 2.13 runs out of heap trying to get some sort of LUB here (I haven't figured out what it's trying to LUB). Ideas/sage advice/sharp insight more than welcome.

Reply:
The whole `A @@ Rep[B]` trick is very rarely used. In general, most cases would use `Injection[A, B]` It may be that something with the tricks to do `@@` 

 https://github.com/twitter/bijection/blob/develop/bijection-core/src/main/scala/com/twitter/bijection/Rep.scala

https://github.com/twitter/bijection/blob/develop/bijection-core/src/main/scala/com/twitter/package.scala#L55